src/akamai_cache_buster/bustCache.py

```diff
@@ -45,32 +45,50 @@ def getHostFromConfig(path="~/.edgerc"):
 # Get the base url using the provided config
 base_url = "https://" + getHostFromConfig()
 
+def createMetadata(path):
+    #create the basic metadata message
+    metadata = '<?xml version=\"1.0\"?>\n<!-- Submitted by bustCache.py script automatically -->\n<eccu>\n'
+
+    #generate the path XML
+    splitPath = path.split('/')
+    metadataClosingTags = ''
+    pathLength = len(splitPath)
+    #create opening and closing tags
+    for i in range(1, pathLength):
+        metadata += '   ' * i + f'<match:recursive-dirs value=\"{splitPath[i]}\">\n'
+        metadataClosingTags += '   ' * (pathLength - i) + '</match:recursive-dirs>\n'
+    metadata += '   ' * pathLength + '<revalidate>now</revalidate>\n'
+    metadata += metadataClosingTags
+    metadata += '</eccu>'
+
+    return metadata
+
+def createRequest(path, appName):
+    body = {
+        "propertyName": "cloud.redhat.com",
+        "propertyNameExactMatch": 'true',
+        "propertyType": "HOST_HEADER",
+        "metadata": createMetadata(path),
+        "notes": "purging cache for new deployment",
+        "requestName": f"Invalidate cache for {appName}",
+        "statusUpdateEmails": [
+            "rfelton@redhat.com"
+        ]
+    }
+
+    return body
 
 #main
 def main():
     appName = sys.argv[2]
-
-    #sys.argv[2] should be name of app to bust
-    print(appName)
     
     #connect to akamai and validate
     initEdgeGridAuth()
 
-    print(getYMLFromUrl("https://cloud.redhat.com/config/main.yml").get(appName).get("frontend").get("paths"))
-
-    urls = []
-    for paths in getYMLFromUrl("https://cloud.redhat.com/config/main.yml").get(appName).get("frontend").get("paths"):
-        urls.append("https://cloud.redhat.com" + paths)
-    
-    for url in urls:
-        print(url)
-
-    body = {
-        "objects" : urls
-    }
-
-    print(base_url + "/ccu/v3/invalidate/url/staging")
-    print(akamaiPost("/ccu/v3/invalidate/url/staging", body))
+    #create a request for each path
+    paths = getYMLFromUrl("https://cloud.redhat.com/config/main.yml").get(appName).get("frontend").get("paths")
+    for path in paths:
+        akamaiPost("/eccu-api/v1/requests", createRequest(path, appName))
```

Comment:
[It looks like it's possible](https://learn.akamai.com/en-us/webhelp/content-control-interfaces/content-control-interfaces/GUID-D045343B-6A5A-4574-B737-F317921C5709.html) however because the paths are various lengths, and we don't necessarily want to clear every possible permutation of the directories that compose the paths we won't be able to follow the best practices they lay out.